core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -164,6 +163,14 @@ class SnappySession(@transient private val sc: SparkContext,
     new SnappySession(sparkContext, Some(sharedState))
   }
 
+  override def createDataFrame[A <: Product : TypeTag](rdd: RDD[A]): DataFrame = {
+    SparkSession.setActiveSession(this)
+    val schema = ScalaReflection.schemaFor[A].dataType.asInstanceOf[StructType]
+    val attributeSeq = schema.toAttributes
+    val rowRDD = RDDConversions.productToRowRdd(rdd, schema.map(_.dataType))
+    Dataset.ofRows(self, LogicalRDD(attributeSeq, rowRDD)(self))
+  }
```

Comment:
Is there an upstream bug for this?

This will not work for other overloads like createDataFrame over java beans, or Seq[A] etc. Also this way is going to be very inefficient compared to the code-generated Encoders. I think better option is to add a separate method (say createDataFrameUsingRDD) instead of override, document the new method for Array[Decimal] and use the same in tests where required for now. Once fix comes upstream then we can remove.